### PR TITLE
feat(speech): add Model Adaptation API

### PR DESCRIPTION
**Pull request by dbolduc**
Part of the work for #12168

- - -
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/12173)

### CODE REVIEW

File: google/cloud/speech/v1/internal/adaptation_logging_decorator.h [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+
+  Status DeleteCustomClass(
+      grpc::ClientContext& context,
+      google::cloud::speech::v1::DeleteCustomClassRequest const& request)
+      override;
+
+ private:
+  std::shared_ptr<AdaptationStub> child_;
+  TracingOptions tracing_options_;
+  std::set<std::string> components_;
```
Comment (devbww) [inline google/cloud/speech/v1/internal/adaptation_logging_decorator.h:91]: You'll need to rebase/regenerate now that I managed to get #12172 in first.

**Review by devbww — APPROVED** [review]

File: google/cloud/speech/v1/internal/adaptation_logging_decorator.h (hunk 1 above)
Comment (dbolduc) [reply google/cloud/speech/v1/internal/adaptation_logging_decorator.h:91]: Yep. And I think we are going to see failures from `retail` soon :upside_down_face:

Comment (devbww) [reply google/cloud/speech/v1/internal/adaptation_logging_decorator.h:91]: I'll fix it up.

Comment (dbolduc) [reply google/cloud/speech/v1/internal/adaptation_logging_decorator.h]: Rebased and regenerated.

**Review by coryan — APPROVED** [review]

**Pull request merged**